### ISSUE TITLE
DL3009 can be ignored when making use of "RUN --mount" optimizations.

### ISSUE THREAD

**Issue by codebymikey** [hadolint/hadolint#943]
- [x] This is a bug report
- [x] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior
There's a way to cache the output of `apt-get` calls through the `RUN` `--mount` option, which should pass validation

References:
* https://stackoverflow.com/a/72851168/21206242
* https://github.com/moby/buildkit/blob/master/frontend/dockerfile/docs/reference.md#run---mounttypecache

### Actual behavior
```bash
$ cat Dockerfile | docker run --rm -i hadolint/hadolint
/dev/stdin:8:39 unexpected "type=cache,s" expecting "from=", "readonly", "readonly=", "readwrite", "readwrite=", "ro", "ro=", "rw", "rw=", source=, or target=
```

### Steps to reproduce the behavior

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```bash
Haskell Dockerfile Linter 2.12.1
```

**Dockerfile** (if relevant)

```Dockerfile
# syntax=docker/dockerfile:1.3.1
FROM ubuntu:jammy

RUN ls -hal /var/lib/apt/lists /etc/apt/apt.conf.d/

RUN cat /etc/apt/apt.conf.d/docker-clean

RUN --mount=target=/var/lib/apt/lists,type=cache,sharing=locked \
    --mount=target=/var/cache/apt,type=cache,sharing=locked \
    rm -f /etc/apt/apt.conf.d/docker-clean \
    && apt-get update \
    && apt-get -y --no-install-recommends install \
        htop \
    && apt-cache madison htop

RUN ls -hal /var/lib/apt/lists /etc/apt/apt.conf.d/

COPY . /tmp/

CMD [ "bash", "-l" ]
```

**Additional environment details (OS, stack version, etc.)**

**Comment by sajith** [comment]
I too came across a similar thing. I have this in my Dockerfile:

```Dockerfile
RUN --mount=source=.git,target=.git,type=bind \
    pip install --no-cache-dir .
```

The lines above result in:

```console
$ hadolint Dockerfile 
Dockerfile:18:37 unexpected "type=bind \<newline>" expecting "from=", "readonly", "readonly=", "readwrite", "readwrite=", "ro", "ro=", "rw", "rw=", source=, or target=
```

I use the bind mount is so that https://github.com/pypa/setuptools_scm would work during `docker build`. It works, but hadolint is unhappy with those lines.

**Comment by rinerb** [comment]
There are tests for mounts in the language-docker library, but I noticed they all had `type=` first.
I moved the `type` to the first position and the error is gone:
```
RUN --mount=type=cache,target=/var/lib/apt/lists,sharing=locked \
    --mount=type=cache,target=/var/cache/apt,sharing=locked \
    rm -f /etc/apt/apt.conf.d/docker-clean \
    && apt-get update \
    && apt-get -y --no-install-recommends install \
        htop \
    && apt-cache madison htop
```

**Comment by m-ildefons** [comment]
Can't reproduce either problem with Hadolint v2.14.0. I believe the underlying issue has been fixed. I'll close this issue.

<img width="1638" height="960" alt="Image" src="https://github.com/user-attachments/assets/ffadf0a2-0787-4eff-98f9-5701307598dc" />
